src/scripts/us-code.test.js

```diff
@@ -0,0 +1,222 @@
+const { getApp, axios } = require("../utils/test");
+
+const usc = require("./us-code");
+
+describe("U.S. Code bot", () => {
+  const app = getApp();
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it("registers a handler", () => {
+    usc(app);
+    expect(app.message).toHaveBeenCalledWith(
+      expect.any(RegExp),
+      expect.any(Function)
+    );
+  });
+
+  describe("handles messages", () => {
+    usc(app);
+    const bot = app.getHandler();
+
+    const say = jest.fn();
+    const message = {
+      context: { matches: [null, 1, 2] },
+      event: { ts: "ts" },
+      message: { text: "" },
+      say,
+    };
+
+    beforeEach(() => {
+      message.message.text = "";
+    });
+
+    describe("if there is any kind of unexpected error", () => {
+      beforeEach(() => {
+        axios.get.mockRejectedValue({});
+      });
+
+      it("does not say anything", async () => {
+        await bot(message);
+        expect(say).not.toHaveBeenCalled();
+      });
+    });
+
+    describe("for a title or subsection that does not exist", () => {
+      beforeEach(() => {
+        axios.get.mockRejectedValue({ response: { status: 404 } });
+      });
+
+      it("tells us it doesn't exist", async () => {
+        await bot(message);
+
+        expect(say).toHaveBeenCalledWith({
+          text: `1 U.S. Code § 2 not found`,
+          thread_ts: "ts",
+        });
+      });
+    });
+
+    describe("for a title and subsection that exists", () => {
+      const source = `
+      <h1 id="page_title">USC Section Name</h1>
+      <div class="section">
+        <div class="content">top-level content</div>
+        <div class="subsection">
+          <span class="num">(a)</span>
+          <div class="content">subsection content</div>
+          <div class="paragraph">
+            <span class="num">(1)</span>
+            <div class="content">paragraph content</div>
+            <div class="subparagraph">
+              <span class="num">(A)</span>
+              <span class="heading">Subparagraph Name</span>
+              <div class="content">subparagraph content</div>
+              <div class="clause">
+                  <span class="num">(i)</span>
+                  <div class="subclause">
+                    <span class="num">(I)</span>
+                    <div class="content">subclause content</div>
+                  </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div class="section">
+`;
+
+      const html = [];
+      beforeEach(() => {
+        html.length = 0;
+        html.push(
+          ...source
+            .trim()
+            .split("\n")
+            .map((v) => v.trim())
+        );
+
+        axios.get.mockImplementation(async () => ({ data: html.join("") }));
+      });
+
+      describe("and the user wants the whole section", () => {
```

Comment:
[comment] you don't need to include the "and" in the test description! the output should do the right thing, i think.

Reply:
I guess it depends on the test runner. I checked in Jest and it doesn't, it just puts a `>` between the describes